### DISCUSSION TITLE
Roon Bridge as Player

### DISCUSSION THREAD

**Discussion by clinds34** [music-assistant/support#1955] [Feature requests and ideas]
I really like what you are doing with MA and HA; I can't beleive I am so late to the party.  I use Roon and specifically Roon Bridge endpoints with SBC/DAC; would be great to add as an option.  

https://download.roonlabs.net/builds/roonbridge-installer-linuxarmv8.sh

**Comment by OzGav** [comment]
What is roon bridge running on?

**Reply by clinds34** [reply]
It is running on a VIM3L SBC (2GB RAM, 16GB eMMC, S905D3) with a headless Ubuntu 22.04.4

**Reply by OzGav** [reply]
Should be able to run squeezelite instead by the looks of it https://manpages.ubuntu.com/manpages/trusty/man1/squeezelite.1.html

**Comment by clinds34** [comment]
thanks - I have added squeezelite and I can see it has recognized.  I was trying to reationalize the # of client (runnning bridge/shairport and now squeeze).  

Appreciate the work and the help.

**Comment by jtatknox** [comment]
You can sort of make this work using squeezelite endpoints or by using the Home Assistant media players introduced by the Roon integration. You can play/pause these Roon Bridge endpoints, but they just play what they were last playing. I do think it would be really really nice to have Roon Bridge.

**Comment by Hedda** [comment]
To clearify, at least as far as I understand there should be no real need to ask for a specific "Roon Bridge Rlayer Provider" dedicated to stream to only Roon Bridges, because "Roon Bridge" is an external official bridging service from Roon that once the user setup as a standalone Roon Bridge application should be able to stream to using the existing [Squeezelite player provider](https://www.music-assistant.io/player-support/squeezelite/) in Music Assistant:

* https://help.roonlabs.com/portal/en/kb/articles/roonbridge

Room Bridge is a standalone headless service (deamon on Linux/Mac) that user install on a other computer running Linux, Mac, or Windows OS and that Room Bridge service will inidrectly expose Roon devices on the network as a Roon Audio Device to which you can stream to using the [Squeezelite player provider](https://www.music-assistant.io/player-support/squeezelite/) (a.k.a. Squeezebox, a.ka. SliMP3) which the Room Bridge in turn proxy and foward to compatible Roon devices on your network:

* https://roon.app/en/downloads

For reference; the "***RoonLabs music player***" integration for Home Assistant also allow control of [RoonLabs](https://roonlabs.com/) music players from Home Assistant:

* https://www.home-assistant.io/integrations/roon

FYI, it sounds as if Home Assistant's Roon integration is used to control a Roon Server or Roon Optimized Core Kit (ROCK) available in Roon OS:

  * https://github.com/home-assistant/core/tree/dev/homeassistant/components/roon

 "_**This integration uses Roon Core, a Roon application that runs on a machine on your network. Via Roon Core, Home Assistant can control all the Roon music players on your network**._"

* https://help.roonlabs.com/portal/en/kb/articles/roon-optimized-core-kit

More about Roon bridging solutions based on Roon Core here:

* https://github.com/roonlabs

* https://roonlabs.eu

Also see pyRoon python library to interface with the Roon API:

* https://github.com/pavoni/pyroon

  * https://pypi.org/project/roonapi/

PS: Off-topic here but FYI for referece; if you have "Roon Ready" endpoints/clients (players/revievers/speakers) then you might perhaps also be interested in this indirectly related feature request which instead asks for native "player provider" to specifically support Roon's RAAT protocol server-side inside Music Assistant's server (as in without an external third-party Roon bridge solution based on the closed-source Roon Core from Roon Labs): 

* https://github.com/orgs/music-assistant/discussions/5133

Referenced back to this thread from there as the request there is off-topic here because it asks for a native solution not using a bridge or proxy:

> _Off-topic here, but for reference, a possible alternative workaround which I do not think is of interest here because they go against the whole concept of Music Assistant would be to just make use of an external third-party "Roon Bridge" that runs as a standalone daemon which proxy services for bridging RAAT to an existing Roon endpoint so that would allow Music Assistant to do for example audio streaming using a other standard protocol to RAAT endpoints indirectly via that external third-party "Roon Bridge" standalone daemon, regardless if that is based on Roon's closed-souce libaries and propriatory SDK , (meaning that Music Assistant just output as Sendspin, Snapcast, Squeezelite, or AirPlay to an existing external Roon Bridge which acts a a proxy to in turn convert that to RAAT and pipe it on the the actual "Roon Ready" endpoint/client devices). For example there are existing Roon bridge software applications used by LMS (Lyrion Music Server, formerly Logitech Media Server). Also see Roon Bridge -> https://desk.zoho.com/portal/roonlabsllc/kb/articles/roonbridge Or the reverse that is Roon's Squeezebox Setup which works by Roon emulating LMS -> https://help.roonlabs.com/portal/en/kb/articles/squeezebox-setup However for a such bridging solution please instead see this other discussion thread which specifically asks for that:_
> 
> * _https://github.com/orgs/music-assistant/discussions/1955_
